File: inubil-verify-front/src/core/api/token-storage.js
/** Stockage des jetons JWT (access + refresh) — cle localStorage dediee, separee de la session factice AuthContext. */
const ACCESS_TOKEN_KEY = 'inubil_access_token';
const REFRESH_TOKEN_KEY = 'inubil_refresh_token';

export function getAccessToken() {
  return localStorage.getItem(ACCESS_TOKEN_KEY);
}

export function getRefreshToken() {
  return localStorage.getItem(REFRESH_TOKEN_KEY);
}

export function setTokens({ access_token, refresh_token }) {
  localStorage.setItem(ACCESS_TOKEN_KEY, access_token);
  localStorage.setItem(REFRESH_TOKEN_KEY, refresh_token);
}

export function clearTokens() {
  localStorage.removeItem(ACCESS_TOKEN_KEY);
  localStorage.removeItem(REFRESH_TOKEN_KEY);
}

File: mobile/lib/core/auth/token_storage.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

/// Stockage securise des jetons JWT (access + refresh) sur l'appareil.
/// Memes cles logiques que le web (localStorage `inubil_access_token` /
/// `inubil_refresh_token`, voir core/api/token-storage.js) — support
/// physique different (keychain/keystore), meme role.
class TokenStorage {
  TokenStorage._();

  static const _storage = FlutterSecureStorage();
  static const _cleAcces = 'inubil_access_token';
  static const _cleRafraichissement = 'inubil_refresh_token';

  static Future<String?> get accessToken => _storage.read(key: _cleAcces);

  static Future<String?> get refreshToken => _storage.read(key: _cleRafraichissement);

  static Future<void> definirJetons({
    required String accessToken,
    required String refreshToken,
  }) async {
    await _storage.write(key: _cleAcces, value: accessToken);
    await _storage.write(key: _cleRafraichissement, value: refreshToken);
  }

  static Future<void> effacer() async {
    await _storage.delete(key: _cleAcces);
    await _storage.delete(key: _cleRafraichissement);
  }
}
